code/datums/antagonists/datum_vampire.dm

```diff
@@ -0,0 +1,280 @@
+		C.hud_used.vamp_blood_display.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#dd66dd'>[round(usable_blood, 1)]</font></div>"
+	handle_vampire_cloak()
+	if(istype(C.loc, /obj/structure/closet/coffin))
+		C.adjustBruteLoss(-4)
+		C.adjustFireLoss(-4)
+		C.adjustToxLoss(-4)
+		C.adjustOxyLoss(-4)
+		return
+	if(!get_ability(/datum/vampire_passive/full) && istype(get_area(C.loc), /area/chapel))
+		vamp_burn()
+	if(isspaceturf(C.loc))
+		check_sun()
+
+
+/datum/antagonist/vampire/proc/handle_bloodsucking(mob/living/carbon/human/H)
+	draining = H
+	var/mob/living/carbon/human/O = owner.current
+	var/blood = 0
+	var/old_bloodtotal = 0 //used to see if we increased our blood total
+	var/old_bloodusable = 0 //used to see if we increased our blood usable
+	log_attack("[O] ([O.ckey]) bit [H] ([H.ckey]) in the neck")
+	O.visible_message("<span class='danger'>[O] grabs [H]'s neck harshly and sinks in their fangs!</span>", "<span class='danger'>You sink your fangs into [H] and begin to drain their blood.</span>", "<span class='notice'>You hear a soft puncture and a wet sucking noise.</span>")
+	if(!iscarbon(owner))
+		H.LAssailant = null
+	else
+		H.LAssailant = O
+	playsound(O.loc, 'sound/weapons/bite.ogg', 50, 1)
+	while(do_mob(O, H, 50))
+		if(!is_vampire(O))
+			to_chat(O, "<span class='warning'>Your fangs have disappeared!</span>")
+			return
+		old_bloodtotal = total_blood
+		old_bloodusable = usable_blood
+		if(!H.blood_volume)
+			to_chat(O, "<span class='warning'>They've got no blood left to give.</span>")
+			break
+		if(H.stat != DEAD)
+			blood = min(20, H.blood_volume)// if they have less than 20 blood, give them the remnant else they get 20 blood
+			total_blood += blood / 2	//divide by 2 to counted the double suction since removing cloneloss -Melandor0
+			usable_blood += blood / 2
+		else
+			blood = min(5, H.blood_volume)	// The dead only give 5 blood
+			total_blood += blood
+		check_vampire_upgrade()
+		if(old_bloodtotal != total_blood)
+			to_chat(O, "<span class='notice'><b>You have accumulated [total_blood] [total_blood > 1 ? "units" : "unit"] of blood[usable_blood != old_bloodusable ? ", and have [usable_blood] left to use" : ""].</b></span>")
+		H.blood_volume = max(H.blood_volume - 25, 0)
+		if(ishuman(O))
+			O.nutrition = min(O.nutrition + (blood / 2), NUTRITION_LEVEL_WELL_FED)
+		playsound(O.loc, 'sound/items/eatfood.ogg', 40, 1)
+
+	draining = null
+	to_chat(owner, "<span class='notice'>You stop draining [H.name] of blood.</span>")
+
+/datum/antagonist/vampire/proc/force_add_ability(path)
+	var/spell = new path(owner)
+	if(istype(spell, /obj/effect/proc_holder/spell))
+		owner.AddSpell(spell)
+	powers += spell
+
+/datum/antagonist/vampire/proc/get_ability(path)
+	for(var/P in powers)
+		var/datum/power = P
+		if(power.type == path)
+			return power
+	return null
+
+/datum/antagonist/vampire/proc/add_ability(path)
+	if(!get_ability(path))
+		force_add_ability(path)

```

Comment:
why would it be forced by default isn't the point of forcing if normal adding doesn't work..?